mitmproxy/controller.py

```diff
@@ -1,22 +1,58 @@
 from __future__ import absolute_import
 from six.moves import queue
 import threading
+import functools
+import sys
 
-from .exceptions import Kill
+from . import exceptions
 
+Events = frozenset([
+    "clientconnect",
+    "clientdisconnect",
+    "serverconnect",
+    "serverdisconnect",
 
-class Master(object):
+    "tcp_open",
+    "tcp_message",
+    "tcp_error",
+    "tcp_close",
+
+    "request",
+    "response",
+    "responseheaders",
+
+    "next_layer",
+
+    "error",
+    "log",
+])
 
+
+class ControlError(Exception):
```

Comment:
This should be `class ControlException(ProxyException): pass` in `mitmproxy.exceptions`.